src/adapter.rs

```diff
@@ -471,6 +475,134 @@ impl Adapter {
+                    SocketAddr::V6(SocketAddrV6::new(address, port, flow_info, scope_id))
+                }
+                _ => { panic!("Illegal address family {}", address_family); }
+            };
+            let mut wg_peer = WireguardPeer {
+                Flags: peer.Flags as u32,
+                PublicKey: peer.PublicKey,
+                PresharedKey: peer.PresharedKey,
+                PersistentKeepalive: peer.PersistentKeepalive,
+                Endpoint: endpoint,
+                TxBytes: peer.TxBytes,
+                RxBytes: peer.RxBytes,
+                LastHandshake: peer.LastHandshake,
+                AllowedIps: Vec::with_capacity(peer.AllowedIPsCount as usize),
+            };
+            for _ in 0..peer.AllowedIPsCount {
+                // # Safety:
+                // 1. `WireGuardGetConfiguration` writes zero or more `WIREGUARD_ALLOWED_IP`s immediately after the WIREGUARD_PEER we read above.
+                // 2. We rely on Wireguard-NT to specify the number of allowed ips written, and therefore we never read too many times unless Wireguard-NT (wrongly) tells us to
+                let allowed_ip: WIREGUARD_ALLOWED_IP = unsafe { reader.read() };
+                let prefix_length = allowed_ip.Cidr;
+                let allowed_ip = match allowed_ip.AddressFamily as i32 {
+                    winapi::shared::ws2def::AF_INET => {
+                        let octets = unsafe { allowed_ip.Address.V4.S_un.S_un_b };
+                        let address = Ipv4Addr::new(octets.s_b1, octets.s_b2, octets.s_b3, octets.s_b4);
+                        IpNet::V4(Ipv4Net::new(address, prefix_length).expect("prefix is valid"))
+                    }
+                    winapi::shared::ws2def::AF_INET6 => {
+                        let octets = unsafe { allowed_ip.Address.V6.u.Byte };
+                        let address = Ipv6Addr::from(octets);
+                        IpNet::V6(Ipv6Net::new(address, prefix_length).expect("prefix is valid"))
+                    }
+                    _ => { panic!("Illegal address family {}", allowed_ip.AddressFamily); }
+                };
+                wg_peer.AllowedIps.push(allowed_ip);
+            }
+            wg_interface.Peers.push(wg_peer);
+        }
+        wg_interface
+    }
+}
+
+#[derive(Debug)]
+pub struct WireguardPeer {
+    /// Bitwise combination of flags
+    pub Flags: u32,
+    /// Public key, the peer's primary identifier
+    pub PublicKey: [u8; 32usize],
+    /// Preshared key for additional layer of post-quantum resistance
+    pub PresharedKey: [u8; 32usize],
+    /// Seconds interval, or 0 to disable
+    pub PersistentKeepalive: u16,
+    /// Endpoint, with IP address and UDP port number
+    pub Endpoint: SocketAddr,
+    /// Number of bytes transmitted
+    pub TxBytes: u64,
+    /// Number of bytes received
+    pub RxBytes: u64,
+    /// Time of the last handshake, in 100ns intervals since 1601-01-01 UTC
+    pub LastHandshake: u64,
```

Comment:
It would be nice to expose this to users as an instant or some kind of more rusty type. But that's something for another day